### PR TITLE
[WIP] Make interpolated strings frozen

### PR DESCRIPTION
**Pull request by kirs**
This is an early attempt to solve https://github.com/oracle/truffleruby/issues/2303.

### CODE REVIEW

File: src/main/java/org/truffleruby/parser/ast/DStrParseNode.java [hunk 1]
```diff
@@ -39,16 +39,26 @@
...
     }
 
+    public boolean isFrozen() {
+        return frozen;
+    }
+
+    public void setFrozen(boolean frozen) {
+        this.frozen = frozen;
+    }
+
```
Comment (kirs) [inline src/main/java/org/truffleruby/parser/ast/DStrParseNode.java:61]: Same methods exist on StrParseNode - should I somehow extract this?

File: src/main/java/org/truffleruby/parser/parser/ParserConfiguration.java [hunk 2]
```diff
@@ -130,4 +137,9 @@ public boolean isSaveData() {
     public boolean isInlineSource() {
         return inlineSource;
     }
+
+    public boolean freezeInterpolatedStrings() {
+        // Core doesn't work with frozen string literals
```
Comment (eregon) [inline src/main/java/org/truffleruby/parser/parser/ParserConfiguration.java:142]: Should be `Core doesn't work with frozen interpolated strings`

File: src/main/java/org/truffleruby/parser/ast/DStrParseNode.java (hunk 1 above)
Comment (chrisseaton) [reply src/main/java/org/truffleruby/parser/ast/DStrParseNode.java:61]: I wouldn't worry about it - this code we're working is already very badly factored and it needs a big spring-clean really.

**Comment by chrisseaton** [comment]
@kirs I fixed a few things if you look through my commits - then we'll need to squash this as it's a bit fractured - then we can open the PR against Oracle.

**Comment by kirs** [comment]
https://github.com/oracle/truffleruby/pull/2304

**Pull request closed without merging**